### PR TITLE
334 clean up userservice unit tests

### PR DESCRIPTION
**Pull request by JelmenGuhlke**
- remove redundant tests for `AbstractUsersService` (`Long` and `UUID`)
- introduce a domain specific jUnit test API
- handle blank password like empty during `User` creation

### CODE REVIEW

File: essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/service/UserServiceStub.java [hunk 1]
```diff
@@ -40,12 +44,12 @@ protected <T extends RoleService> LongUserService(
...
 
   @Override
-  protected @NotNull <E extends UserDto<Long>> TestLongUser convertDtoToEntity(@NotNull E entity) {
+  public @NotNull <E extends UserDto<Long>> UserStub convertDtoToEntity(@NotNull E entity) {
     HashSet<Role> roles =
         entity.getRoles().stream()
             .map(roleService::getByName)
             .collect(Collectors.toCollection(HashSet::new));
-    return TestLongUser.builder()
+    return UserStub.builder()
```
Comment (pwannenmacher) [inline essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/service/UserServiceStub.java:52]: Isn't a `.loginDisabled(entity.isLoginDisabled())` missing here?

File: essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/service/UserServiceStubUUID.java [hunk 2]
```diff
@@ -41,7 +45,7 @@ protected <T extends RoleService> UUIDUserService(
   }
 
   @Override
-  protected @NotNull <E extends UserDto<UUID>> TestUUIDUser convertDtoToEntity(@NotNull E entity) {
+  public @NotNull <E extends UserDto<UUID>> TestUUIDUser convertDtoToEntity(@NotNull E entity) {
```
Comment (pwannenmacher) [inline essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/service/UserServiceStubUUID.java:48]: Isn't a `.loginDisabled(entity.isLoginDisabled())` missing here?

File: essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/user/TestObjectsUser.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        .id(TEST_USER_ID)
+        .email(TEST_USERNAME)
+        .enabled(true)
+        .firstName(TEST_FIRST_NAME)
+        .lastName(TEST_LAST_NAME)
+        .locale(TEST_LOCALE)
+        .password(TEST_PASSWORD_HASH)
+        .passwordResetToken(null)
+        .nonce(TEST_NONCE)
+        .source(TEST_SOURCE)
```
Comment (pwannenmacher) [inline essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/user/TestObjectsUser.java:35]: Isn't a `.loginDisabled(_???_)` missing here?

File: essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/annotations/EssenciumUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * The EssenciumUnitTest annotation should be used on all jUnit test classes in the Essencium
+ * project. It enables the jUnit extensions {@link TestObjectInjectionExtension} and {@link
+ * MetricCleanUpExtension}, which provides utility functions to create test objects and interact
+ * with mocks. Especially the {@link TestObjectInjectionExtension} enables the possibility to create
+ * objects via parameter injection. See {@link
+ * TestObjectInjectionExtension#resolveParameter(ParameterContext, ExtensionContext)} for more
+ * information.
+ */
+@ExtendWith(TestObjectInjectionExtension.class)
```
Comment (pwannenmacher) [inline essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/annotations/EssenciumUnitTest.java:22]: Couldn't the annotation `@ExtendWith(MockitoExtension.class)` also be included here and we would save ourselves the trouble of adding it to every test class?

File: essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/service/UserServiceStub.java (hunk 1 above)
Comment (JelmenGuhlke) [reply essencium-backend/src/test/java/de/frachtwerk/essencium/backend/api/data/service/UserServiceStub.java:60]: Good catch - it was missing already before the PR changes. I will add it. As I said - using a Stub and not a "real" productive used object is only testing the test implementation. There is no guarantee, that the prod implementation is the same.

**Review by pwannenmacher — APPROVED** [review]

**Pull request merged**